console/console-init/ui/src/modules/address-space/containers/AddressSpaceFilterContainer/AddressSpaceFilterContainer.tsx

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import * as React from "react";
+import { DataToolbarChip } from "@patternfly/react-core/dist/js/experimental";
+import { SelectOption, SelectOptionObject } from "@patternfly/react-core";
+import { RETURN_ALL_ADDRESS_SPACES_FOR_NAME_OR_NAMESPACE } from "graphql-module/queries";
+import { ISearchNameOrNameSpaceAddressSpaceListResponse } from "types/ResponseTypes";
+import { useApolloClient } from "@apollo/react-hooks";
+import { FetchPolicy, TYPEAHEAD_REQUIRED_LENGTH } from "constants/constants";
+import { ISelectOption, getSelectOptionList } from "utils";
+import {
+  AddressSpaceFilter,
+  AddressSpaceListKebab
+} from "modules/address-space/components";
+
+interface IAddressSpaceFilterContainerProps {
+  filterValue?: string;
+  setFilterValue: (value: string) => void;
+  filterNames: any[];
+  setFilterNames: (value: Array<any>) => void;
+  filterNamespaces: any[];
+  setFilterNamespaces: (value: Array<any>) => void;
+  filterType?: string | null;
+  setFilterType: (value: string | null) => void;
+  totalAddressSpaces: number;
+}
+
+export const AddressSpaceFilterContainer: React.FunctionComponent<IAddressSpaceFilterContainerProps> = ({
+  filterValue,
+  setFilterValue,
+  filterNames,
+  setFilterNames,
+  filterNamespaces,
+  setFilterNamespaces,
+  filterType,
+  setFilterType,
+  totalAddressSpaces
+}) => {
+  const client = useApolloClient();
+  const [filterIsExpanded, setFilterIsExpanded] = React.useState<boolean>(
+    false
+  );
+  const [typeFilterIsExpanded, setTypeFilterIsExpanded] = React.useState<
+    boolean
+  >(false);
+  const [isSelectNameExpanded, setIsSelectNameExpanded] = React.useState<
+    boolean
+  >(false);
+  const [
+    isSelectNamespaceExpanded,
+    setIsSelectNamespaceExpanded
+  ] = React.useState<boolean>(false);
+  const [nameSelected, setNameSelected] = React.useState<string>();
+  const [namespaceSelected, setNamespaceSelected] = React.useState<string>();
+  const [nameOptions, setNameOptions] = React.useState<Array<ISelectOption>>();
+  const [nameInput, setNameInput] = React.useState<string>("");
+  const [nameSpaceInput, setNameSpaceInput] = React.useState<string>("");
+  const [namespaceOptions, setNamespaceOptions] = React.useState<
+    Array<ISelectOption>
+  >();
+
+  const filterMenuItems = [
```

Comment:
I think we should declare the type dropdownItems as IDropdownOption[ ] (already defined)